### PR TITLE
committing spec for romanTranslatorService

### PR DESCRIPTION
**Pull request by Aish1206**
Hello,
Please review the specs for romanNumeralTranslator Service.

### CODE REVIEW

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js [hunk 1]
```diff
@@ -1,19 +1,115 @@
...
   beforeEach(module('myApp.components.romanNumbers'));
 
   var service;
+  var mapping;
 
-  beforeEach(inject(function(romanNumeralTranslatorService) {
-
-    service = romanNumeralTranslatorService;
+  beforeEach(inject(function(_romanNumeralTranslatorService_) {
+    service = _romanNumeralTranslatorService_;
```
Comment (kburson) [inline app/components/romanNumbers/romanNumeralTranslator.service.spec.js:12]: ok, completely unnecessary, but it shows you understand about the injector and the '_' wrapping of dependencies.

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js [hunk 2]
```diff
@@ -1,19 +1,115 @@
...
+      };
+
+      function checkIt(output, input){
+        it('should return Roman Number ' + output + ' when input is a number with value ' + input, function(){
+          var returnValue = service.translate(input);
+          expect(returnValue).toEqual(output);      
+        });
+      }
+
+      angular.forEach(mapping, function(output, input){
```
Comment (kburson) [inline app/components/romanNumbers/romanNumeralTranslator.service.spec.js:38]: impressive, straight up data driven boundary conditions.  What about edge cases (either side of boundaries)?

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js [hunk 3]
```diff
@@ -1,19 +1,115 @@
...
+        checkIt(output, input);
+      });
+
+    });
+
+    it('should return Roman Number XII when input is a string with value 12', function() {
```
Comment (kburson) [inline app/components/romanNumbers/romanNumeralTranslator.service.spec.js:44]: Why 12 and not 11?

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js (hunk 2 above)
Comment (kburson) [reply app/components/romanNumbers/romanNumeralTranslator.service.spec.js:38]: What about numbers 1..10 ?

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js [hunk 4]
```diff
@@ -1,19 +1,115 @@
...
+      var returnValue = service.translate('12');
+      expect(returnValue).toEqual('XII');
+    });
+
+    it('should not return anything when input is a number with value -2 which is less than 0', function() {
```
Comment (kburson) [inline app/components/romanNumbers/romanNumeralTranslator.service.spec.js:49]: simple if number is < 0, no need to point out 'number 2'.

What about '0' ?

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js [hunk 5]
```diff
@@ -1,19 +1,115 @@
...
+      var returnValue = service.translate(-2);
+      expect(returnValue).toEqual('');
+    });
+
+    it('should return Roman Number I when input is a number with value 1 which is less than 11 and greater than 0', function() {
+      var returnValue = service.translate(1);
+      expect(returnValue).toEqual('I');
+    });
 
+    it('should return Roman Number XII when input is a number with value 12 which is less than 40 and greater than 10', function() {
```
Comment (kburson) [inline app/components/romanNumbers/romanNumeralTranslator.service.spec.js:59]: you already tested for 12

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js [hunk 6]
```diff
@@ -1,19 +1,115 @@
...
+      var returnValue = service.translate(12);
+      expect(returnValue).toEqual('XII');
+    });
 
+    it('should return Roman Number XLV when input is a number with value 45 which is less than 50 greater than 39', function() {
```
Comment (kburson) [inline app/components/romanNumbers/romanNumeralTranslator.service.spec.js:64]: you are making it obvious that you understand boundary conditions, this is good, but you do not need to spell them out in the test description, only the behavior.

'should return 'XLV' for input of '45'

File: karma.conf.js [hunk 7]
```diff
@@ -32,7 +32,7 @@ module.exports = function (config) {
 
     preprocessors: {
       '**/*.html': 'ng-html2js',
-      'app/**/!(*.spec).js': ['coverage']
+      '**/!(*.spec).js': ['coverage']
```
Comment (kburson) [inline karma.conf.js:35]: interesting, what other files are you expecting to capture with this?

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js [hunk 8]
```diff
@@ -1,19 +1,115 @@
...
+      expect(returnValue).toEqual('CMXCV');
+    });
+
+    it('should return Roman Number MXCV when input is a number with value 1095 which is less than 3888 greater than 999', function() {
+      var returnValue = service.translate(1095);
+      expect(returnValue).toEqual('MXCV');
+    });
+
+    it('should throw an error when value is greater than 3888', function() {
+      expect(function(){service.translate(3990)}).toThrow(new Error('invalid number, max value = 3888'));
```
Comment (kburson) [inline app/components/romanNumbers/romanNumeralTranslator.service.spec.js:110]: why 3990 and not 3889 ?

File: app/components/romanNumbers/romanNumeralTranslator.service.spec.js [hunk 9]
```diff
@@ -1,19 +1,115 @@
...
+    it('should return Roman Number XII when input is a number with value 12 which is less than 40 and greater than 10', function() {
+      var returnValue = service.translate(12);
+      expect(returnValue).toEqual('XII');
+    });
 
+    it('should return Roman Number XLV when input is a number with value 45 which is less than 50 greater than 39', function() {
+      var returnValue = service.translate(45);
+      expect(returnValue).toEqual('XLV');
+    });
+
```
Comment (kburson) [inline app/components/romanNumbers/romanNumeralTranslator.service.spec.js:68]: what about the 4's and 9's ? every 4 and 9 creates a swap of digits:
3,4,5,6 == III, **IV**, V, **VI**      13,14,15,16 == XIII, XIV, XV, **XVI**
8,9,10,11 == VIII, **IX**, X, **XI**   18,19,20,21 == XVII, XIX, XX, **XXI**

Comment (kburson) [reply app/components/romanNumbers/romanNumeralTranslator.service.spec.js:68]: And similarly when you approach other break points (50, 100, 500 ...)
